src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -4662,6 +4713,166 @@ public void CopyCtor_InaccessibleInMetadata()
                 );
         }
 
+        [Fact, WorkItem(45077, "https://github.com/dotnet/roslyn/issues/45077")]
+        public void CopyCtor_AmbiguitiesInMetadata()
+        {
+            // IL for a minimal `public record B { }` with injected copy constructors
+            var ilSource_template = @"
+.class public auto ansi beforefieldinit B extends [mscorlib]System.Object
+{
+    INJECT
+
+    .method public hidebysig specialname newslot virtual instance class B '<>Clone' () cil managed
+    {
+        IL_0000: ldarg.0
+        IL_0001: newobj instance void B::.ctor(class B)
+        IL_0006: ret
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Object::.ctor()
+        IL_0006: ret
+    }
+}
+";
+
+            var source = @"
```

Comment:
>source [](start = 16, length = 6)

This variable is used only in one place. Consider inlining, I think this will make the test code easier to follow